src/app/frontend/logs/component.js

```diff
@@ -92,8 +96,19 @@ export class LogsController {
     /** @private {!ui.router.$stateParams} */
     this.stateParams_;
 
-    /** @export {!kdUiRouter.$transition$} - initialized from resolve */
+    /** @export {!kdUiRouter.$transition$} */
     this.$transition$;
+
+    /** @type {boolean} */
+    this.isFollowing = false;
+
+    // TODO(maciaszczykm): Add time interval to settings page.
+    this.interval_(() => {
+      if (this.isFollowing) {
+        // TODO Add log.
+        this.loadNewest();
+      }
+    }, 5000);
```

Comment:
Actually, I had implemented this and @floreks did not like it :)

Anyway, I would reduce the time to at least one second. In addition it would be possible to increase the performance and and reduce the load of the apiserver by limiting the api call to 100 lines.

Reply:
True :), I don't like polling very much and would prefer exposing watch endpoint directly from apiserver. It would allow "live refresh" without causing too much load on our backend and on apiserver.

If we proceed with that as a temporary solution I'd rather keep refresh number rather high (5s+) not to cause too much load. We can make this option global when settings page will be introduced and give option to choose from few predefined intervals.

Reply:
I was not aware. I consider is a temporary solution too. First I have started with just a refresh button, but this is a bit better and we had few requests for it. Next step will be adding refresh interval to the settings page, then we can think about a more clever solution.

Reply:
The watch semantic seems not to be fully correct. If a user presses the refresh button we need not only all updates, but also all updates that happend between loading and pressing buttion. Anyway, the API provides a follow flag that keeps the stream open. I was thinking to pass this stream to the frontend. However, to make this work it would require to return a stream of json objects (or text directly) to the frontend instead of one. 

From UX perspective I am not fully convinced if 5s or settings page is reasonable. Lets imagine a use case:  You are devops, start a server and wait for a log output... I would be bothered with 5s. I guess in such a case a refresh button without auto-refresh is better. So a user just click multiple times himself to refresh. Certainly not perfect, but maybe least cumbersome.

Reply:
I think both make sense, but I would keep this minimal auto-refresh at the moment. It makes first call to the API on click and then every 5 secs, so the user should be able to see what happens. I will add time interval value to the tooltip, so it will be better to understand.

Reply:
> The watch semantic seems not to be fully correct. If a user presses the refresh button we need not only all updates, but also all updates that happend between loading and pressing button.

In our case since we are using pagination, we do not really care about data in the middle during refresh. Refresh means for us "load last logs page and show it to the user". 

> From UX perspective I am not fully convinced if 5s or settings page is reasonable. Lets imagine a use case: You are devops, start a server and wait for a log output... I would be bothered with 5s.

As a dev I'm more concerned about putting too much load on backend and apiserver by decreasing this number. From UX perspective it is still an improvement since there is no option to auto-refresh logs right now. We need some compromise between performance and usability. Even though it might not be perfect for containers that produce logs very quickly, it will still be ok in many cases.

> I guess in such a case a refresh button without auto-refresh is better. So a user just click multiple times himself to refresh. Certainly not perfect, but maybe least cumbersome.

Refresh normally loads last portion of logs and button to do this is already there so adding yet another button that does the same thing is not that nice. User can simply click on last page button multiple times and see how logs are "refreshing". That's what I'm usually doing.

Reply:
>In our case since we are using pagination, we do not really care about data in the middle during refresh. Refresh means for us "load last logs page and show it to the user".

Long term, clicking the button should do 'refresh + follow'. Luckily, the api server is exactly doing that. We just have to refactor and expose a similar behavior to the front end.